winsup/cygwin/mm/malloc_wrapper.cc

```diff
@@ -318,9 +331,9 @@ malloc_init ()
       extern void *_sigfe_malloc;
       /* Decide if we are using our own version of malloc by testing the import
 	 address from user_data.  */
-      use_internal = true; // user_data->malloc == malloc
-//		     || import_address ((void *) user_data->malloc)
-//			== &_sigfe_malloc;
+      use_internal = user_data->malloc == malloc
+		     || import_address ((void *) user_data->malloc)
+			== malloc;
```

Comment:
A proper solution would keep the original code here.

Reply:
Yes, it is a workaround which should be reverted after fixing _sigfe_malloc.

Reply:
And what does it actually improve? E.g., what does not work without this change?

Reply:
It reverts some changes and port import_address to aarch64.

Reply:
I'd like to understand what it improves from functional prespective? E.g., what was failing before and does not fail now?

Reply:
I do not have a repro, it was a result of the investigation. However it is one more step forward to keep the original code and understand sigfe issue.